osquery/main/windows/daemon.cpp

```diff
@@ -54,7 +66,124 @@ void DebugPrintf(const char *fmt, ...) {
   va_end(vl);
 }
 
-void UpdateServiceStatus(DWORD controls, DWORD state, DWORD exit_code, DWORD checkpoint) {
+/// Install osqueryd as a service given the path to the binary
+Status installOsqueryService(const std::string& binPath) {
```

Comment:
You're implicitly creating a r-value `std::string` then later only accessing the `char*` internal representation. If you pass a `const char* const binPath` you'll save the conversion.